src/NextGen/DonationForm/DataTransferObjects/DonationFormViewRouteData.php

```diff
@@ -34,11 +34,39 @@ public static function fromRequest(array $request): DonationFormViewRouteData
         $self = new static();
 
         $self->formId = (int)$request['form-id'];
-        $self->formSettings = !empty($request['form-settings']) ? $request['form-settings'] : [];
+        $self->formSettings = !empty($request['form-settings']) ? $self->filterFormSettings(
+            $request['form-settings']
+        ) : [];
         $self->formBlocks = !empty($request['form-blocks']) ? BlockCollection::fromJson(
             $request['form-blocks']
         ) : null;
 
         return $self;
     }
+
+    /**
+     * The route is receiving boolean params as strings.
+     * This is a temporary hot fix to find those values and cast them to booleans.
+     * I'm sure there is a better way to do this.
+     *
+     * @unreleased
+     */
+    public function filterFormSettings(array $settings): array
+    {
+        return array_map(static function ($setting) {
+            if ($setting === 'false') {
+                return false;
+            }
+
+            if ($setting === 'true') {
+                return true;
+            }
+
+            if (filter_var($setting, FILTER_VALIDATE_BOOLEAN)) {
+                return (bool)$setting;
+            }
+
+            return $setting;
```

Comment:
You're kind of reinventing `filter_var`, here. This whole thing could be:
```php
return filter_var($setting, FILTER_VALIDATE_BOOLEAN);`
```

If you want to prevent false as default on error, you can do:
```php
$filteredValue = filter_var($setting, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE);

if ( $filteredValue === null ) {
  throw new InvalidArgumentException('Invalid setting, must be boolean. Received ' . $setting);
}

return $filteredValue;
```

This will handle ["true" and "false" as values](https://www.php.net/manual/en/filter.filters.validate.php).

Reply:
I think i'm just going to check specifically for the words `true` and `false` and get rid of filter_var.  It's specific to those and don't want this thinking string numbers are booleans.

Reply:
Does the system know which settings to anticipate as booleans?

Reply:
resolved up the chain 04e7e109